dataclients/kubernetes/crd.md

```diff
@@ -0,0 +1,634 @@
+  algorithm: <string> optional, only user for type=lb
+  endpoints: <stringarray> required for type=lb
+  serviceName: <string> required for type=service
+  servicePort: <number> required for type=service
+  default: <bool> optional  // TODO: maybe separate object
+  weight: <number> optional // TODO: maybe separate object
+```
+
+`<stringarray>` is an array of `<string>`
+`<number>` is an unsigned integer, example: 80
+`<bool>` is a either `true` or `false`
+
+```
+<route>
+  path: <string>  // TODO: maybe separate object and rest put into object
+  exactPath: <bool> optional // TODO: does it makes sense to have this or should this be the default?
+  backend: <string>, optional references a specific <backend>
+  filters: <stringarray>, optional
+  predicates: <stringarray>, optional
+  method: <string>, optional, one of the HTTP methods "GET|HEAD|PATCH|POST|PUT|DELETE|CONNECT|OPTIONS|TRACE", defaults to all // TODO maybe indent into an object and move filters,predicates,backends to it
+```
+
+### RouteGroup expressed use cases
+
+#### Complex route with redirect and migration
+
+Assume we have a Kubernetes service `my-service-v1` which gets all
+requests for hostnames www.complex.example.org and
+complex.example.org.
+
+We might want to specify a redirect to our login service.
+
+Maybe some day we want to migrate from /api to /, in our backend. Old services should be able to use /api, which they can later change to /.
+
+```yaml
+apiVersion: zalando.org/v1
+kind: RouteGroup
+metadata:
+  name: my-routes
+spec:
+  hosts:
+  - www.complex.example.org
+  - complex.example.org
+  backends:
+  - name: my-service
+    type: service
+    serviceName: my-service-v1
+    servicePort: 80
+    default: true
+  - name: redirect
+    type: shunt
+  - name: migration
+    type: loopback
+  routes:
+  - path: /
+  - path: /api
+    backend: migration
+    filters:
+    - modPath("/api", "/")
+  - path: /login
+    method: get
+    backend: redirect
+    filters:
+    - redirectTo(308, "https://login.example.org/)
+```
+
+
+#### Complex routes with ratelimits based on tokens
+
+A complex route case could specify different ratelimits for POST
+request to /api/resource for clients.  Clients that have JWT/OAuth2
+Tokens from issuer https://accounts.google.com with email
+"important@example.org" get 20/equest per minute, other clients with
+Token with issuer https://accounts.google.com or
+https://accounts.github.com get only 2.
+
+Additionally for all other HTTP methods ratelimit for each client 100
+requests per minute based on "Authorization" header and make sure the
+Token is valid and has the scopes to read and list the resource.
+
+```yaml
+apiVersion: zalando.org/v1
+kind: RouteGroup
+metadata:
+  name: api
+spec:
+  hosts:
+  - api.example.org
+  - api.service.example.net
+  backends:
+  - name: api-svc
+    type: service
+    serviceName: api-service-v1
+    servicePort: 80
+    default: true
+  routes:
+  - path: /api/resource
+    method: post
```

Comment:
This should be a list. It's highly unlikely that someone would want to setup different policies for e.g. `put` vs `post` vs `patch`, so I'd rather cover the 99% case in something that wouldn't require copy-pasting the route.

Reply:
So you mean, that it's likely to have the same rules for the `writes` and others for the `reads` like different scopes in auth filters, for example